stan/math/rev/fun/positive_ordered_constrain.hpp

```diff
@@ -85,7 +22,37 @@ class positive_ordered_constrain_op {
  */
 inline Eigen::Matrix<var, Eigen::Dynamic, 1> positive_ordered_constrain(
     const Eigen::Matrix<var, Eigen::Dynamic, 1>& x) {
-  return adj_jac_apply<internal::positive_ordered_constrain_op>(x);
+  using std::exp;
+
+  size_t N = x.size();
+  if (N == 0) {
+    return x;
+  }
+
+  Eigen::VectorXd y_val(N);
+  arena_matrix<Eigen::VectorXd> exp_x(N);
+
+  exp_x(0) = exp(value_of(x.coeffRef(0)));
+  y_val(0) = exp_x.coeffRef(0);
+  for (int n = 1; n < N; ++n) {
+    exp_x(n) = exp(value_of(x.coeffRef(n)));
```

Comment:
```suggestion
    exp_x.coeffRef(n) = exp(value_of(x.coeff(n)));
```
Also use coeffRef for writing values. For reading coeff is a bit shorter.